### PR TITLE
Rename barclamp nova_dashboard to horizon

### PR DESCRIPTION
**Pull request by vwallfahrer**
Depends on https://github.com/SUSE-Cloud/automation/pull/531

### CODE REVIEW

File: bin/crowbar_nova_dashboard [hunk 1]
```diff
@@ -17,7 +17,7 @@
 #
 
 require File.join(File.expand_path(File.dirname(__FILE__)), "barclamp_lib")
-@barclamp = "nova_dashboard"
+@barclamp = "horizon"
```
Comment (vuntz) [inline bin/crowbar_nova_dashboard]: Shouldn't the file be renamed to crowbar_horizon?

File: chef/cookbooks/horizon/recipes/ha.rb [hunk 2]
```diff
@@ -1,4 +1,4 @@
-# Copyright 2014 SUSE
+%# Copyright 2014 SUSE
```
Comment (vuntz) [inline chef/cookbooks/horizon/recipes/ha.rb]: I guess that character wasn't intended?

File: chef/roles/horizon-server.rb [hunk 3]
```diff
@@ -0,0 +1,8 @@
+name "horizon-server"
+description "Nova Dashboard Server Role"
```
Comment (vuntz) [inline chef/roles/horizon-server.rb]: Maybe use Horizon in the string?

File: crowbar_framework/app/models/horizon.rb [hunk 4]
```diff
@@ -43,22 +43,22 @@ def role_constraints
...
-    answer << { "barclamp" => "nova", "inst" => role.default_attributes["nova_dashboard"]["nova_instance"] }
+    answer << { "barclamp" => "database", "inst" => role.default_attributes["horizon"]["database_instance"] }
+    answer << { "barclamp" => "keystone", "inst" => role.default_attributes["horizon"]["keystone_instance"] }
+    answer << { "barclamp" => "nova", "inst" => role.default_attributes["horizon"]["nova_instance"] }
     answer
   end
 
   def create_proposal
-    @logger.debug("Nova_dashboard create_proposal: entering")
+    @logger.debug("horizon create_proposal: entering")
```
Comment (vuntz) [inline crowbar_framework/app/models/horizon.rb]: s/horizon/Horizon/

File: crowbar_framework/app/models/horizon.rb [hunk 5]
```diff
@@ -68,26 +68,26 @@ def create_proposal
 
     base["attributes"][@bc_name][:db][:password] = random_password
 
-    @logger.debug("Nova_dashboard create_proposal: exiting")
+    @logger.debug("horizon create_proposal: exiting")
```
Comment (vuntz) [inline crowbar_framework/app/models/horizon.rb]: Same here.

**Comment by tboerger** [comment]
LGTM

File: crowbar_framework/app/models/horizon.rb [hunk 6]
```diff
@@ -68,26 +68,26 @@ def create_proposal
...
   def validate_proposal_after_save proposal
-    validate_one_for_role proposal, "nova_dashboard-server"
+    validate_one_for_role proposal, "horizon-server"
 
     super
   end
 
   def apply_role_pre_chef_call(old_role, role, all_nodes)
-    @logger.debug("Nova_dashboard apply_role_pre_chef_call: entering #{all_nodes.inspect}")
+    @logger.debug("horizon apply_role_pre_chef_call: entering #{all_nodes.inspect}")
```
Comment (vuntz) [inline crowbar_framework/app/models/horizon.rb]: Sorry, didn't mention in the previous review, but uppercase H would be nice here.

File: crowbar_framework/app/models/horizon.rb [hunk 7]
```diff
@@ -149,7 +149,7 @@ def apply_role_pre_chef_call(old_role, role, all_nodes)
       node.save
     end
 
-    @logger.debug("Nova_dashboard apply_role_pre_chef_call: leaving")
+    @logger.debug("horizon apply_role_pre_chef_call: leaving")
```
Comment (vuntz) [inline crowbar_framework/app/models/horizon.rb]: Same here.

**Comment by vuntz** [comment]
Can you also update the files in crowbar_framework/examples/ (they got added after you started working in this PR)?

**Comment by dirkmueller** [comment]
How do we plan to do do the migration/updating for existing customers? I am lacking to find the code that takes care of migration.

**Comment by rsalevsky** [comment]
The migration is part of the `Create process for upgrading admin server from Cloud 5 to Cloud 6` https://trello.com/c/v1m4Ay1c card.

**Comment by rsalevsky** [comment]
Something is wrong with the commit, maybe wrong rebase?

**Comment by rsalevsky** [comment]
Renaming of `chef/data_bags/crowbar/bc-template-nova_dashboard.json` and `chef/data_bags/crowbar/bc-template-nova_dashboard.schema` are missing.

**Comment by rsalevsky** [comment]
LGTM but needs some manual testing because it depends on another PR

**Comment by vuntz** [comment]
See https://github.com/SUSE-Cloud/automation/pull/531#issuecomment-139989390 -- seemed that this should work fine if we merge both, so going ahead.

**Pull request merged**